### ISSUE TITLE
Syntax for using modules as dependencies in other modules

### ISSUE THREAD

**Issue by emilsoman** [fika-lang/fika#29]
Syntax will look like: `use <path_relative_to_root> as <alias>`.
Examples: `use fika/parser as Parser`, `use deps/my_library as MyLibrary`.

We already infer module names from their filesystem paths. This is similar to [how Gleam modules work](https://gleam.run/book/tour/modules.html).

One important detail that's worth discussing:
Should we use snake_case for module names or CamelCase? If we use snake_case, we can allow users to just write `use foo/my_module` instead of `use foo/my_module as Parser`. But the advantage of using CamelCase for module names is that it's easier on the eyes for people coming from Elixir. One could ask why not automatically create CamelCase module names instead of asking the user and the answer is - we want to make it explicit.

Why `use` instead of `alias`? A new programmer should be able to understand what "using" a module means, also it's shorter to type. Why do we not import functions? When looking at a function call, the user should be able to easily trace back where it's defined. This means, do not import other modules' functions into the current module so they look like local functions.

The compiler will throw an error if a user calls a function `Foo.bar` without first mentioning `use <module path> as Foo`

**Comment by polvalente** [comment]
Just to leave some thoughts here:

- `use` vs `alias` vs `import` doesn't really make a difference in my opinion. We have `#include` in C, `import` and `from ... import` (which do different things) in Python, `use`/`alias`/`import` in Elixir and so many other variations, that I think that each language imbues their own meaning to these terms. Hence, anyone is good for me!

- I guess that **requiring** `use` or `use ... as` would open up for accepting any one (or both?) of PascalCase or snake_case, since my initial thoughts about confusion between modules and variables would be nullified, since no module would be auto-used in the current module besides the kernel module. Also, the point:
> the advantage of using CamelCase for module names is that it's easier on the eyes for people coming from Elixir

I don't necessarily think that's true because a) There are lots of dynamic modules around (see `Ecto.Multi.run`, that passes a dynamic repo to be optionally used); b) Erlang interop required snake_case.

- Finally, one thing I thought as I was typing this is that in both _cases_ (pun intended), we could borrow from Python and have submodule aliasing with `from ... use .. as`, which could help in situations with really long paths:

For a file structure like:
```
my/
  really/
    really/
      really/
        nested/
          module.fi
          module/
            foo.fi
            bar/
              baz.fi
```

```
use my/really/really/really/nested/module as nested_module
from nested_module use foo as foo
from nested_module use bar/baz as baz
```

Or maybe we just accept paths as strings, and then aliases like this could just be accomplished as simple string concatenation before the `use ... as` directive. Like so:


```
# List used as a placeholder
nested_path = "my/really/really/really/nested/module"
use [nested_path, "foo"] as foo
use [nested_path, "bar/baz"] as baz
```

Or we could even introduce another keyword for these base module substitutions

**Comment by emilsoman** [comment]
Closed by #51
